src/radical/pilot/umgr/staging_input/default.py

```diff
@@ -325,26 +325,58 @@ def _handle_unit(self, unit, actionables):
         self._prof.prof("create_sandbox_stop", uid=uid)
 
         # Loop over all transfer directives and execute them.
+        removables = list()
         for sd in actionables:
 
             action = sd['action']
             flags  = sd['flags']
             did    = sd['uid']
             src    = sd['source']
             tgt    = sd['target']
+            if action == rpc.TARBALL:
+                # If the staging method is tarball, check initially if the tar 
+                # file exists. If not create it. Add each file in the tarball
+                # and append the directive to the removables.
 
-            self._prof.prof('staging_in_start', uid=uid, msg=did)
+                if not os.path.isfile(uid+'.tar'):
+                    tar_file = tarfile.open(uid+'.tar','w')
+                tar_file.add(src.split('client:///')[1],arcname=tgt.split('unit:///')[1])
```

Comment:
Are you *sure* that the src and tgt specs match that pattern here?  Maybe use URL parsing and then `url.path`?  This differs from how the `elif` branch handles the URLs.